backend/src/main/java/codezap/template/domain/Template.java

```diff
@@ -65,16 +65,61 @@ public class Template extends SkipModifiedAtBaseTimeEntity {
     @Enumerated(EnumType.STRING)
     private Visibility visibility;
 
-    public Template(Member member, String title, String description, Category category, Visibility visibility) {
-        this(null, member, title, description, category, null, 0L, visibility);
+    @Column(nullable = false)
+    @ColumnDefault("0")
+    private Long thumbnailOrdinal;
+
+    @ElementCollection(fetch = FetchType.LAZY)
+    @OrderColumn(name = "ordinal")
+    @CollectionTable(name = "source_code", joinColumns = @JoinColumn(name = "template_id"))
+    private final List<SourceCode> sourceCodes = new ArrayList<>();
+
+    public Template(
+            Long id,
+            Member member,
+            String title,
+            String description,
+            Category category,
+            Long likesCount,
+            Visibility visibility,
+            Long thumbnailOrdinal,
+            List<SourceCode> sourceCodes
+    ) {
+        this.id = id;
+        this.member = member;
+        this.title = title;
+        this.description = description;
+        this.category = category;
+        this.likesCount = likesCount;
+        this.visibility = visibility;
+        this.thumbnailOrdinal = thumbnailOrdinal;
+        this.sourceCodes.addAll(sourceCodes);
+        validateSourceCodeCount(sourceCodes);
+    }
+
+    public Template(Member member, String title, String description, Category category, Visibility visibility, List<SourceCode> sourceCodes) {
+        this(null, member, title, description, category, 0L, visibility, 0L, sourceCodes);
     }
 
-    public void updateTemplate(String title, String description, Category category, Visibility visibility) {
+    public void updateTemplate(String title, String description, Category category, Visibility visibility, List<SourceCode> sourceCodes) {
         this.modifiedAt = LocalDateTime.now();
         this.title = title;
         this.description = description;
-        this.category = category;
         this.visibility = visibility;
+        this.category = category;
+        recreateSourceCodes(sourceCodes);
+        validateSourceCodeCount(sourceCodes);
+    }
+
+    private void validateSourceCodeCount(List<SourceCode> sourceCodes) {
+        if (sourceCodes.isEmpty()) {
+            throw new CodeZapException(ErrorCode.INVALID_REQUEST, "소스 코드는 최소 1개 입력해야 합니다.");
+        }
+    }
+
+    private void recreateSourceCodes(List<SourceCode> sourceCodes) {
```

Comment:
```suggestion
    private void updateSourceCodes(List<SourceCode> sourceCodes) {
```
굳이 코드 내부에서 삭제하고 다시 만든다는걸 밖으로 드러낼 필요는 없어보여요~~

Reply:
private 메서드라 Template에서는 노출되는데 오히려 요구사항이 명확하게 보인다고 생각했어요~!
외부에서 직접 접근이 가능하다면 몰라도, 노출하지 않는 것이 좋지만 오히려 템플릿 도메인만을 본다고 생각하면 재생성하는 것이 낫다고 생각하는데 어떤가용??

Reply:
저는 "Template를 업데이트" 하는 과정에서 "SourceCode를 업데이트 한다" 라는 단계가 포함되는 것이 추상화 수준에서 자연스럽다고 생각해서 남긴 의견이에요.

만약 지금처럼 `SourceCode`를 삭제 후 재생성하는 것 보다 단순히 내용을 업데이트하는게 성능이 더 좋다는 것을 알게되었다고 가정해 보면..
저였다면 `updateSourceCodes()` 메서드의 상세 구현을 재생성 -> 단순 업데이트 로 변경했을 것 같아서요.
결국 더 높은 수준의 추상화단계인 `updateTemplate()`에서는 여전히 소스코드를 업데이트 하고 있을 뿐 변경되는 게 없을거고요.

하지만, 몰리가 생각하기에 "SourceCode를 **재생성 한다**" 라는 사실이 어느정도 중요한 의미를 담고 있고, 상위 메서드에서도 이를 나타내는 것이 더 좋다고 생각한다면 기존의 메서드명도 나쁘지는 않아보입니다~!